### ISSUE TITLE
Background scan stops

### ISSUE THREAD

**Issue by mash** [opentrace-community/opentrace-ios#18]
This is a theoretical issue, I haven't tested this but should happen.

Here you're trying to periodically start and stop scanning.
https://github.com/opentrace-community/opentrace-ios/blob/master/OpenTrace/Bluetrace/CentralController.swift#L103

When the app goes background, right after stopScan is called, background scanning will not work.

See https://developer.apple.com/library/archive/documentation/NetworkingInternetWeb/Conceptual/CoreBluetooth_concepts/CoreBluetoothBackgroundProcessingForIOSApps/PerformingTasksWhileYourAppIsInTheBackground.html#//apple_ref/doc/uid/TP40013257-CH7-SW9

> The services the central manager was scanning for (and any scan options specified when the scan started)

Was this, scanning start/stop dance proved to be effective ?
As I understand iOS can be scanning and advertising at the same time without a problem.

**Comment by erwanmace** [comment]
@mash you may want to check this other issue #4

**Comment by wabzqem** [comment]
I have confirmed that this happens with the COVIDSafe application using this code:
https://docs.google.com/document/d/1sviVey1jCk97BACwkAJMctG8rFlC7Mf8pMpluW6VOQw/preview

This could definitely work much better.

**Comment by mash** [comment]
Agree. I tried the background task approach that was proposed in #4, and it worked. I think that's the max we can get out of the current iOS. I'll update my own repo https://github.com/mash/BLE-Proximity with my findings. There seems no activity here...
